addons/binding/org.openhab.binding.http/src/main/java/org/openhab/binding/http/internal/HttpHandlerFactory.java

```diff
@@ -0,0 +1,113 @@
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.smarthome.config.core.Configuration;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandlerFactory;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandlerFactory;
+import org.eclipse.smarthome.core.thing.link.ItemChannelLinkRegistry;
+import org.eclipse.smarthome.io.net.http.HttpClientFactory;
+import org.openhab.binding.http.handler.HttpThingHandler;
+import org.openhab.binding.http.model.HttpHandlerFactoryConfig;
+import org.osgi.service.component.ComponentContext;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+import java.util.Dictionary;
+import java.util.Enumeration;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.openhab.binding.http.HttpBindingConstants.THING_TYPE_HTTP;
+
+/**
+ * Handler factory for creating handlers for things of type http.
+ *
+ * @author Brian J. Tarricone
+ */
+@NonNullByDefault
+@Component(service = ThingHandlerFactory.class, configurationPid = "binding.http")
+public class HttpHandlerFactory extends BaseThingHandlerFactory {
+    private final Map<ThingUID, HttpThingHandler> handlers = new HashMap<>();
+
+    private HttpClient httpClient;
+    private ItemChannelLinkRegistry itemChannelLinkRegistry;
+    private HttpHandlerFactoryConfig config = new HttpHandlerFactoryConfig();
+
+    @Override
+    protected ThingHandler createHandler(final Thing thing) {
+        if (supportsThingType(thing.getThingTypeUID())) {
+            final HttpThingHandler handler = new HttpThingHandler(
+                    thing, this.httpClient, this.itemChannelLinkRegistry,
+                    this.config.getConnectTimeout(), this.config.getRequestTimeout()
+            );
+            this.handlers.put(thing.getUID(), handler);
+            return handler;
+        } else {
+            return null;
+        }
+    }
+
+    @Override
+    public boolean supportsThingType(final ThingTypeUID thingTypeUID) {
+        return THING_TYPE_HTTP.equals(thingTypeUID);
+    }
+
+    @Override
+    public void unregisterHandler(final Thing thing) {
+        super.unregisterHandler(thing);
+        this.handlers.remove(thing.getUID());
+    }
+
+    @Override
+    protected void activate(final ComponentContext componentContext) {
+        super.activate(componentContext);
+        final Map<String, Object> properties = dictionaryToMap(componentContext.getProperties());
+        this.config = new Configuration(properties).as(HttpHandlerFactoryConfig.class);
+        this.handlers.forEach((uid, handler) -> handler.updateBindingConfig(this.config.getConnectTimeout(), this.config.getRequestTimeout()));
```

Comment:
I see, you hand the config to your handlers. Why not make it a handler configuration in the first place and save yourself those round-trips. Because you have not defined a `modified` method
```
 @Modified
  public void modified(Map<String, Object> params)
```
OSGI will also terminate (`deactivate`) your factory (including all its handlers) on a configuration change and call `activate` again. You really might be better of having this Thing/Channel configurations.

Reply:
Yeah that makes sense...  moved them to channel config, since people might need different timeouts depending on server/endpoint/etc.  Then I won't need this at all in the factory and can delete some code...